### PR TITLE
Add documentation for CLI commands

### LINKED ISSUE #38: Create documentation for CLI command
Create documentation for the `wp plugin check` command. This documentation will live in this repo.

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/WordPress/plugin-check/issues/38

### CODE REVIEW

File: docs/wp-plugin-check.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+[Back to overview](./README.md)
+
+# wp plugin check
```
Comment (swissspidy) [inline docs/wp-plugin-check.md]: Can we try using `wp scaffold package-readme` for this so the documentation is automated?

See https://github.com/wp-cli/scaffold-package-command#wp-scaffold-package-readme

Comment (ernilambar) [reply docs/wp-plugin-check.md]: Sure. I will do research.

Comment (ernilambar) [reply docs/wp-plugin-check.md]: `wp scaffold package-readme` requires valid package directory as parameter with valid `composer.json`. So as of now we would not be able to use that command in our repo. When we separate our CLI and make it as a package then only we would be able to use this.

Comment (swissspidy) [reply docs/wp-plugin-check.md]: What updates to composer.json do we need to make?

Comment (ernilambar) [reply docs/wp-plugin-check.md]: I dug deeper into the scaffold command but faced multiple issues. To generate doc content it used `wp cli cli-dump` commands output but there all commands are there but our `plugin check` is not there. Also location of `README.md` and also file name is not configurable. I am testing in this PR https://github.com/WordPress/plugin-check/pull/356

Comment (swissspidy) [reply docs/wp-plugin-check.md]: Ah, I see, right, it looks at all the registered commands, but the ones in the plugin are of course only registered when WP is active. Too bad!

Then this should be OK for now.

**Comment by ernilambar** [comment]
I developed simple helper CLI command to generate doc for the CLI commands. https://github.com/ernilambar/pc-cli-docs So doc generation is now semi-automatic instead of manual.
It would be great if we could find a way incorporate this command in the trunk and exclude it when we deploy to SVN in the plugin directory.

Preview: https://github.com/ernilambar/plugin-check/blob/38-add-cli-documentation/docs/CLI.md

There is no `EXAMPLES` section for `wp plugin list-checks` because it is missing in the command itself. I have created separate PR to add example. https://github.com/WordPress/plugin-check/pull/396

**Comment by swissspidy** [comment]
It would be better to contribute to https://github.com/wp-cli/scaffold-package-command to make it easier to generate documentation for commands within a plugin. Kudos for building something custom. However, long term it is not viable to fork and maintain a custom tool that just works with Plugin Check. Especially since these commands don't change frequently.

**Review by swissspidy — APPROVED** [review]

File: docs/CLI.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+By default, `.git`, `vendor` and `node_modules` directories are excluded.
+```
+## EXAMPLES
+```
+wp plugin check akismet
+wp plugin check akismet --checks=late_escaping
+wp plugin check akismet --format=json
+```
+
+# wp plugin list-checks
```
Comment (mukeshpanchal27) [inline docs/CLI.md:55]: Add `EXAMPLES` that added in #396

File: docs/CLI.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  - csv
+  - json
+---
+```
+## EXAMPLES
+```
+wp plugin list-check-categories
+wp plugin list-check-categories --format=json
+```
+
```
Comment (mukeshpanchal27) [inline docs/CLI.md]:
```suggestion
```

**Review by mukeshpanchal27 — COMMENTED** [review]
Thanks for the PR! I've left some minor feedback for your consideration.

File: docs/CLI.md (hunk 3 above)
Comment (ernilambar) [reply docs/CLI.md]: Done

**Review by mukeshpanchal27 — APPROVED** [review]

**Pull request merged**